### PR TITLE
docs: add master key provider examples

### PR DESCRIPTION
**Pull request by mattsb42-aws**
*Issue #, if available:*
#213
#146

*Description of changes:*

To aid in migration efforts, these examples show the basics of how to use master key providers in the new example format.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

# Check any applicable:
- [ ] Were any files moved? Moving files changes their URL, which breaks all hyperlinks to the files.

### CODE REVIEW

**Comment by mattsb42-aws** [comment]
Note for the record: all failing checks are expected. They are all from the test vector handler workflow. The tests from that workflow actually should not be running on pull request events, but I'll figure that out in a different PR.

File: examples/src/master_key_provider/aws_kms/discovery_decrypt.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+"""
+This example is provided as a reference for users migrating away from master key providers.
+We recommend that all new use should use keyrings.
```
Comment (WesleyRosenblum) [inline examples/src/master_key_provider/aws_kms/discovery_decrypt.py:5]: this wording is a little awkward

File: examples/src/master_key_provider/aws_kms/discovery_decrypt.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+The KMS master key provider uses any key IDs that you specify on encrypt,
+but attempts to decrypt *any* data keys that were encrypted under a KMS CMK.
+This means that you do not need to know which CMKs were used to encrypt a message.
+
+This example shows how to configure and use a KMS master key provider to decrypt without provider key IDs.
+
+https://docs.aws.amazon.com/encryption-sdk/latest/developer-guide/concepts.html#master-key-provider
+
+For an example of how to use the KMS master key provider with CMKs in multiple regions,
+see the ``master_key_provider/aws_kms/multiple_regions`` example.
```
Comment (WesleyRosenblum) [inline examples/src/master_key_provider/aws_kms/discovery_decrypt.py:17]:
```suggestion
For an example of how to use the KMS master key with a single CMK,
see the ``master_key_provider/aws_kms/single_cmk`` example.

For an example of how to use the KMS master key provider with CMKs in multiple regions,
see the ``master_key_provider/aws_kms/multiple_regions`` example.
```

File: examples/src/master_key_provider/aws_kms/multiple_regions.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+This example shows how to configure and use a KMS master key provider with with CMKs in multiple regions.
+
+https://docs.aws.amazon.com/encryption-sdk/latest/developer-guide/concepts.html#master-key-provider
+
+For an example of how to use the KMS master key with a single CMK,
+see the ``master_key_provider/aws_kms/single_cmk`` example.
+
+For examples of how to use the KMS master key provider in discovery mode on decrypt,
+see the ``master_key_provider/aws_kms/discovery_decrypt``.
```
Comment (WesleyRosenblum) [inline examples/src/master_key_provider/aws_kms/multiple_regions.py:16]:
```suggestion
For an example of how to use the KMS master key provider in discovery mode on decrypt,
see the ``master_key_provider/aws_kms/discovery_decrypt`` example.
```

File: examples/src/master_key_provider/aws_kms/single_cmk.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+This example shows how to configure and use a KMS master key with a single KMS CMK.
+
+https://docs.aws.amazon.com/encryption-sdk/latest/developer-guide/concepts.html#master-key-provider
+
+For an example of how to use the KMS master key provider with CMKs in multiple regions,
+see the ``master_key_provider/aws_kms/multiple_regions`` example.
+
+For examples of how to use the KMS master key provider in discovery mode on decrypt,
+see the ``master_key_provider/aws_kms/discovery_decrypt``.
```
Comment (WesleyRosenblum) [inline examples/src/master_key_provider/aws_kms/single_cmk.py:16]:
```suggestion
For an example of how to use the KMS master key provider in discovery mode on decrypt,
see the ``master_key_provider/aws_kms/discovery_decrypt`` example.
```

**Pull request merged**